crates/system/os/transaction/src/batch_of_transactions/apply_shields/apply_shield_transactions_builder/helpers/poly_manifest_builder/tx_fee_modifier.rs

```diff
@@ -0,0 +1,104 @@
+use crate::prelude::*;
+
+pub(super) trait ApplyShieldTransactionsManifestTxFeeModifier:
+    Send + Sync
+{
+    fn add_fee_for_unsecurified_account(
+        &self,
+        input: ApplicationInputForUnsecurifiedAccount,
+    ) -> Result<ApplicationInputForUnsecurifiedAccount>;
+
+    fn add_fee_for_unsecurified_persona(
+        &self,
+        input: ApplicationInputForUnsecurifiedPersona,
+    ) -> Result<ApplicationInputForUnsecurifiedPersona>;
+
+    fn add_fee_for_securified_persona(
+        &self,
+        input: ApplicationInputForSecurifiedPersona,
+        _manifest_variant: RolesExercisableInTransactionManifestCombination,
+    ) -> Result<ApplicationInputForSecurifiedPersona>;
+
+    fn add_fee_securified_account(
+        &self,
+        input: ApplicationInputForSecurifiedAccount,
+        _manifest_variant: RolesExercisableInTransactionManifestCombination,
+    ) -> Result<ApplicationInputForSecurifiedAccount>;
+}
+
+pub(super) struct ApplyShieldTransactionsManifestTxFeeModifierImpl {}
+impl Default for ApplyShieldTransactionsManifestTxFeeModifierImpl {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl ApplyShieldTransactionsManifestTxFeeModifierImpl {
+    pub fn new() -> Self {
+        Self {}
+    }
+}
+
+impl ApplyShieldTransactionsManifestTxFeeModifier
```

Comment:
are these 4 correct @GhenadieVP  ?

Reply:
not it is not correct... bah! Updating now.

Reply:
Changed `add_fee_for_unsecurified_account` to lock fee against other fee payer - if specified.